tests/deploy/test_scheduler_agent.py

```diff
@@ -2263,12 +2328,40 @@ async def test_state_of_skipped_resources_for_dependencies(agent: TestAgent, mak
     executor2.deploys[rid2].set_result(const.HandlerResourceState.skipped_for_dependency)
     await retry_limited_fast(utils.is_agent_done, scheduler=agent.scheduler, agent_name="agent2")
 
+    assert agent.scheduler._state.resource_state[rid1] == state.ResourceState(
+        status=state.ComplianceStatus.NON_COMPLIANT,
+        deployment_result=state.DeploymentResult.FAILED,
+        blocked=state.BlockedStatus.NO,
+    )
+
     assert agent.scheduler._state.resource_state[rid2] == state.ResourceState(
         # We are skipped, so not compliant
         status=state.ComplianceStatus.NON_COMPLIANT,
         deployment_result=state.DeploymentResult.SKIPPED,
         blocked=state.BlockedStatus.TRANSIENT,
     )
+    # Release a new version where r1 succeeds and assert that r2 also gets deployed
```

Comment:
Aren't the changes to this test case completely redundant? As in, covered by the new test case you added?